### PR TITLE
fix #252: handle keyboard interrupts

### LINKED ISSUE #252: Handle KeyboardInterrupts when using `MDRunner.run()`
Sending a `Ctrl-C` interrupt during an mdrun kills the program but, as far as I can tell, leaves the simulation running with no easy way to kill it. I think that, ideally, interrupting the program should also kill background simulations.

### PR DESCRIPTION
**Pull request by jandom**
I have again paired with my coding robot and managed to hammer out some nice fix – plus a test!

### CODE REVIEW

File: tests/test_run.py [hunk 1]
```diff
@@ -54,6 +58,49 @@ def test_MDRunner():
...
+    # Initialize MDrunner
+    mdrunner = gromacs.run.MDrunner()
+
+    # Function to send SIGINT after a delay
+    def send_interrupt():
+        time.sleep(1)  # Short delay before sending SIGINT
+        os.kill(os.getpid(), signal.SIGINT)
+
+    # Start a thread to send the SIGINT
+    interrupt_thread = threading.Thread(target=send_interrupt)
```
Comment (orbeckst) [inline tests/test_run.py:87]: That is nifty/hacky. Did the robot suggest it?

**Review by orbeckst — CHANGES_REQUESTED** [review]
Given that tests are happy, just add an entry to CHANGES.

You and your 🤖  are on a role. Maybe I have to update my priors regarding 🤖s.

**Comment by jandom** [comment]
The robot did indeed suggest it but it took a lot of prompting :)

File: tests/test_run.py (hunk 1 above)
Comment (jandom) [reply tests/test_run.py:87]: The robot did indeed suggest it but it took a lot of prompting :)

**Comment by jandom** [comment]
All changed now @orbeckst

**Review by orbeckst — APPROVED** [review]

**Pull request merged**